### PR TITLE
Support 2-Decimal Formatting & customizable decimal separator

### LINKED ISSUE #173: 2-decimal formatting?
Please answer these question so I can help you faster.

**Is it a bug or a feature request?**

- [ ] It's a bug.
- [X] I'm requesting a new feature.

**Which version are you using?**

- [ ] macOS
- [X] Windows
- [X] Linux

**Which is the version/distribution of the OS?**
I have been using Manta 1.1.1 on a Windows 10 laptop and on a Linux Mint 18.2 Sonya laptop.

**Describe your issue**
First of all, I must compliment you on a superb application. I have only one request: When certain purchase amounts are entered (such as $25.00 and $12.50 for instance), the values are printed as $25 and $12.5.  Is it possible to maintain a 2 decimal format for values in the invoices?

Thanks for your consideration. I really like this application.

**Steps to replicate**
[None]

### LINKED ISSUE #190: Customizable decimal separator
**Is it a bug or a feature request?**

- [ ] It's a bug.
- [x] I'm requesting a new feature.

**Which version are you using?**

- [ ] macOS
- [x] Windows
- [x] Linux

**Which is the version/distribution of the OS?**

Windows: Windows 10 Fall Creators Update
Linux: Ubuntu 17.10

**Describe your issue**

Most european countries use a comma (,) as decimal separator and not a dot (.).
It will be great to have a setting to choose the decimal separator.

**Steps to replicate**
- Enter amount on invoice
- Print / Preview invoice

**Remarks**
I will submit a PR this week to implement this.

### PR DESCRIPTION
**Pull request by DarkSmile92**
Shows 2 fractions on decimals and makes the decimal separator region aware.

Close #190, Closes #173

### CODE REVIEW

**Comment by hql287** [comment]
@DarkSmile92: This is not finished yet, right?

**Comment by DarkSmile92** [comment]
@hql287 No I am adding the two settings right now and implement the display of currency sign before / after the amount. Will commit to PR and let you know once I'm done.
the latest tomorrow, have a lot of meetings today, sorry :(

**Comment by hql287** [comment]
@DarkSmile92 No pressure, Robin. I just add [WIP] to the title, feel free to remove it once you're done with the PR. Thanks!

**Comment by DarkSmile92** [comment]
Thanks @hql287 !
If you have time, please review the current changes. I think I got all relevant places but maybe I missed some places.

**Comment by hql287** [comment]
@DarkSmile92 Will do. Thanks!

**Comment by hql287** [comment]
@DarkSmile92 Robin, I just tested out your PR and this is pretty cool. Everything seems to be working as expected. I really like the currency sign placement setting. Nice touch! 👍

There're still some issues though:

* Did you forget about #190? I don't see the setting to choose between comma (,) and dot (.)
* If the price is `499.50` and the fraction is 2, it will print `499` which is incorrect. 
* If the decimal fraction is negative (eg -2), it will return an error. I know this is silly but it would be nice to show a dialogue rather than letting the app crash. 
* The tests are broken

**Comment by DarkSmile92** [comment]
@hql287 Hung you are right, I totally forgot #190 sorry :worried:
I will check the other points and edit the tests (which I also forgot to customize).
I'll commit tonight

**Comment by hql287** [comment]
@DarkSmile92 Thanks, bro! 👍

**Comment by DarkSmile92** [comment]
- [x] #190 Settings for decimal separator
- [x] Checked with price of `499.50`. Shows correct fractions when using `499.50`.
- [x] Catched exception when fraction setting is <0 (with message for user)
I customized the settings components to have a state "canSave". This can be used for further validations on the settings later.
It throws a warning notification and does not let the user save if decimalFractions is < 0
- [x] Tests customized

File: app/actions/settings.jsx [hunk 1]
```diff
@@ -17,3 +17,9 @@ export const saveSettings = createAction(
   ACTION_TYPES.SETTINGS_SAVE,
   data => data
 );
+
+// Notify user about invalid decimal fractions
+export const notifyInvalidDecimalFractions = createAction(
```
Comment (hql287) [inline app/actions/settings.jsx]: Please use `dialog` for validation message. I would like to user to acknowledge this before moving on.

**Comment by hql287** [comment]
@DarkSmile92 Robin, [can you allow me to push to your PR](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/)?

File: app/actions/settings.jsx (hunk 1 above)
Comment (DarkSmile92) [reply app/actions/settings.jsx]: Changed :)

**Comment by DarkSmile92** [comment]
@hql287 Already checked, you should be able to push to my PR Hung 👍

**Comment by hql287** [comment]
@DarkSmile92 I just added some new definitions to facilitate the changes in this PR. Can you do the translation for those term on Crowdin (I would like to add German support with this PR)? 

Thanks!

**Comment by DarkSmile92** [comment]
@hql287 Sure, translation on Crowdin done :)

**Comment by hql287** [comment]
Almost done. 

- [ ] Add validation rule (and tests) for currency in Form/Setting
- [ ] Add test for formatNumber helper
- [ ] Add migration sheme for exsiting invoice

@DarkSmile92: If you can review the changes, that would be great! 👍

**Comment by DarkSmile92** [comment]
@hql287 sorry for not responding, something came up at work that needed my full attention. I reviewed the changes. I really like that you made currency an object with decimal and placement properties. This makes it extensible. All changes look good to me to I agree. Can't approve them via git review so take this as review succeeded and approved:)

**Comment by hql287** [comment]
@DarkSmile92 Thanks, bro 👍

**Pull request merged**